tests/Meilisearch.Tests/IndexTests.cs

```diff
@@ -9,11 +10,179 @@ namespace Meilisearch.Tests
     public class IndexTests
     {
         private MeilisearchClient defaultClient;
+        private string defaultPrimaryKey;
 
         public IndexTests(IndexFixture fixture)
         {
             fixture.DeleteAllIndexes().Wait(); // Test context cleaned for each [Fact]
             this.defaultClient = fixture.DefaultClient;
+            this.defaultPrimaryKey = "movieId";
+        }
+
+        [Fact]
+        public async Task BasicIndexCreation()
+        {
+            var indexUid = "BasicIndexCreationTest";
+            var index = await this.defaultClient.CreateIndex(indexUid);
+            index.Uid.Should().Be(indexUid);
+            index.PrimaryKey.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task IndexCreationWithPrimaryKey()
+        {
+            var indexUid = "IndexCreationWithPrimaryKeyTest";
+            var index = await this.defaultClient.CreateIndex(indexUid, this.defaultPrimaryKey);
+            index.Uid.Should().Be(indexUid);
+            index.PrimaryKey.Should().Be(this.defaultPrimaryKey);
+        }
+
+        [Fact]
+        public async Task BasicUsageOfIndexMethod()
+        {
+            var indexUid = "BasicUsageOfIndexMethodTest";
+            var index = this.defaultClient.Index(indexUid);
+            index.Uid.Should().Be(indexUid);
+            index.PrimaryKey.Should().BeNull();
+            MeilisearchApiError ex = await Assert.ThrowsAsync<MeilisearchApiError>(() => this.defaultClient.GetIndex(indexUid));
+            Assert.Equal("index_not_found", ex.ErrorCode);
+        }
+
+        [Fact]
+        public async Task ImplicitIndexCreationWithPrimaryKey()
+        {
+            var indexUid = "ImplicitIndexCreationWithPrimaryKeyTest";
+            var index = this.defaultClient.Index(indexUid);
+            index.Uid.Should().Be(indexUid);
+            var document = await index.AddDocuments(new[] { new Movie { Id = "1", Name = "Batman" } }, this.defaultPrimaryKey);
+            index.PrimaryKey.Should().BeNull();
```

Comment:
Here you don't test an implicit index creation because you don't test the index is created.
You should:
- use a primary key that does not contain `id` in the name (otherwise MeiliSearch is able to infer)
- wait for the document addition to end
- check you can get the index and the primary key is the good one
- do this test of implicit index creation with a document addition without primary key

Anyway I'm not sure this test should be added in the IndexTests.cs file, it's as you wish.

Reply:
I deleted it